File: PreDemObjcDemo/PreDemObjcDemo/ViewController.m
//
//  ViewController.m
//  PreDemObjcDemo
//
//  Created by WangSiyu on 21/02/2017.
//  Copyright © 2017 pre-engineering. All rights reserved.
//

#import "ViewController.h"
#import <PreDemObjc/PREDemObjc.h>


@interface ViewController ()
<
UIPickerViewDataSource,
UIPickerViewDelegate
>

@property (nonatomic, strong) IBOutlet UILabel *versionLable;
@property (nonatomic, strong) IBOutlet UIPickerView *logLevelPicker;
@property (nonatomic, strong) NSArray *logPickerKeys;
@property (nonatomic, strong) NSArray *logPickerValues;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    self.versionLable.text = [NSString stringWithFormat:@"%@(%@)", PREDManager.version, PREDManager.build];
    self.logLevelPicker.dataSource = self;
    self.logLevelPicker.delegate = self;
    self.logPickerKeys = @[
                               @"不上传 log",
                               @"PREDLogLevelOff",
                               @"PREDLogLevelError",
                               @"PREDLogLevelWarning",
                               @"PREDLogLevelInfo",
                               @"PREDLogLevelDebug",
                               @"PREDLogLevelVerbose",
                               @"PREDLogLevelAll"
                               ];
    self.logPickerValues = @[
                           @(PREDLogLevelOff),
                           @(PREDLogLevelError),
                           @(PREDLogLevelWarning),
                           @(PREDLogLevelInfo),
                           @(PREDLogLevelDebug),
                           @(PREDLogLevelVerbose),
                           @(PREDLogLevelAll)
                           ];
}

- (IBAction)logTest:(id)sender {
    PREDLogVerbose(@"verbose log test");
    PREDLogDebug(@"debug log test");
    PREDLogInfo(@"info log test");
    PREDLogWarn(@"warn log test");
    PREDLogError(@"error log test");
}

- (IBAction)sendHTTPRequest:(id)sender {
    NSArray *urls = @[@"http://www.baidu.com", @"https://www.163.com", @"http://www.qq.com", @"https://www.dehenglalala.com", @"http://www.balabalabalatest.com", @"http://www.alipay.com"];
    for (NSString *urlString in urls) {
        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), ^{
            NSURL *url = [NSURL URLWithString:urlString];
            NSURLRequest *request = [NSURLRequest requestWithURL:url];
            NSURLSession *session = [NSURLSession sharedSession];
            NSURLSessionDataTask *task = [session dataTaskWithRequest:request];
            [task resume];
        });
    }
}

- (IBAction)diagnoseNetwork:(id)sender {
    [PREDManager  diagnose:@"www.qiniu.com"
                  complete:^(PREDNetDiagResult * _Nonnull result) {
        NSLog(@"new diagnose completed with result:\n %@", result);
    }];
}

- (IBAction)forceCrash:(id)sender {
    @throw [NSException exceptionWithName:@"Manually Exception" reason:@"嗯，我是故意的" userInfo:nil];
}

- (IBAction)diyEvent:(id)sender {
    NSDictionary *dict = @{
                           @"stringKey": [NSString stringWithFormat:@"test\t_\n%d", arc4random_uniform(100)],
                           @"longKey": @(arc4random_uniform(100)),
                           @"floatKey": @(arc4random_uniform(10000)/100.0)
                           };
    PREDEvent *event = [PREDEvent eventWithName:@"test\t_\nios\t_\nevent_2" contentDic:dict];
    [PREDManager trackEvent:event];
}

- (IBAction)blockMainThread:(id)sender {
    sleep(1);
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

// returns the number of 'columns' to display.
- (NSInteger)numberOfComponentsInPickerView:(UIPickerView *)pickerView {
    return 1;
}

// returns the # of rows in each component..
- (NSInteger)pickerView:(UIPickerView *)pickerView numberOfRowsInComponent:(NSInteger)component {
    return self.logPickerKeys.count;
}

- (nullable NSString *)pickerView:(UIPickerView *)pickerView titleForRow:(NSInteger)row forComponent:(NSInteger)component __TVOS_PROHIBITED {
    return self.logPickerKeys[row];
}

- (void)pickerView:(UIPickerView *)pickerView didSelectRow:(NSInteger)row inComponent:(NSInteger)component __TVOS_PROHIBITED; {
    if (row == 0) {
        [PREDLogger stopCaptureLog];
    } else {
        [PREDLogger startCaptureLogWithLevel:(PREDLogLevel)[self.logPickerValues[row-1] intValue]];
    }
}


@end

File: PreDemObjcDemo/PreDemSwiftDemo/ViewController.swift
//
//  ViewController.swift
//  PreDemSwiftDemo
//
//  Created by 王思宇 on 05/09/2017.
//  Copyright © 2017 pre-engineering. All rights reserved.
//

import UIKit
import PreDemObjc

class ViewController: UIViewController, UIPickerViewDataSource, UIPickerViewDelegate {
    
    enum CustomError: Error {
        case CustomError(String)
    }
    
    @IBOutlet var versionLable: UILabel!
    @IBOutlet var logLevelPicker: UIPickerView!
    
    let logPickerKeys = [
        "不上传 log",
        "PREDLogLevelOff",
        "PREDLogLevelError",
        "PREDLogLevelWarning",
        "PREDLogLevelInfo",
        "PREDLogLevelDebug",
        "PREDLogLevelVerbose",
        "PREDLogLevelAll"
    ]
    let logPickerValues = [
        PREDLogLevel.off,
        PREDLogLevel.error,
        PREDLogLevel.warning,
        PREDLogLevel.info,
        PREDLogLevel.debug,
        PREDLogLevel.verbose,
        PREDLogLevel.all,
        ]
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func sendHttpRequest(sender: Any) {
        let urls = [
            "http://www.baidu.com",
            "https://www.163.com",
            "http://www.qq.com",
            "https://www.dehenglalala.com",
            "http://www.balabalabalatest.com",
            "http://www.alipay.com",
            ]
        for urlString in urls {
            if let url = URL(string: urlString) {
                URLSession.shared.dataTask(with: URLRequest.init(url: url)).resume()
            } else {
                print("url not valid \(urlString)")
            }
        }
    }
    
    @IBAction func blockMainThread(sender: Any) {
        sleep(1)
    }
    
    @IBAction func forceCrash(sender: Any) {
        try!{throw CustomError.CustomError("嗯，我是故意的")}()
    }
    
    @IBAction func diagnoseNetwork(sender: Any) {
        PREDManager.diagnose("www.qiniu.com") { (result) in
            print("new diagnose completed with result:\n\(result)")
        }
    }
    
    @IBAction func diyEvent(sender: Any) {
        if let event = PREDEvent(name: "viewDidLoadEvent", contentDic: ["helloKey": "worldValue", "hellonum": 7]) {
            PREDManager.trackEvent(event)
        }
    }
    
    @IBAction func logTest(sender: Any) {
        PREDLogVerbose("verbose log test");
        PREDLogDebug("debug log test");
        PREDLogInfo("info log test");
        PREDLogWarn("warn log test");
        PREDLogError("error log test");
    }
    
    func numberOfComponents(in pickerView: UIPickerView) -> Int {
        return 1
    }
    
    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
        return logPickerKeys.count
    }
    
    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
        return logPickerKeys[row]
    }
    
    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
        if row == 0 {
            PREDLogger.stopCaptureLog()
        } else {
            PREDLogger.startCaptureLog(with: logPickerValues[row-1])
        }
    }
    
    
}

